MRN-Code/coinstac
# Simulator Cleanup

- **Problem:** _coinstac-simulator_ can only run ~4 clients (see #27 for details). This appears to be due to a race conditioning where client processes can’t get an active consortium document from the database.
- **Solution:** Make client processes retry consortium database requests 4 times to avoid race condition (see 60b893576cb6b9fcf6d8ece4be7dd2926dd9d892).
- **Bonus:** This refactors some simulator code:
  - Move database server booting from a spawned process to into main process. It’s using pouchdb-server, which already spawns a new child process.
  - Makes configuration sensible.
  - Moves CLI to _bin_ directory (should we canonize this for all projects?)
- **Testing:**
  1. Check out the _bug/#27-simulator-clients_ branch
  2. Clone [swashcap/coinstac-simulator-load-tester](https://github.com/swashcap/coinstac-simulator-load-tester)
  3. Link _coinstac-simulator_ globally and link it to _coinstac-simulator-load-tester_.
  4. Run `node src/index.js -c 10` inside the _-load-tester_ to simulate 10 clients.
